docs/architecture/edge_service_specification.md

```diff
@@ -50,6 +50,8 @@ required group attributes {
   optional string x_pingsender_version // example: "1.0"
   optional string x_debug_id           // example: "my_debug_session_1"
   optional string x_pipeline_proxy     // time that the AWS->GCP tee received the message, example: "2018-03-12T21:02:18.123456Z"
+  optional string x_ecosystem_anon_id  // Anonymized identifier for Account Ecosystem Telemetry
+  optional string x_prev_ecosystem_anon_id  // Previous anon_id value; sent when rotating identifiers
```

Comment:
I need to get more detail on the size of these `anon_id` payloads. The maximum size that a `PubsubMessage` supports per attribute value is 1024 bytes. If `anon_id` needs to be larger than this, than we have to encode it in the payload. We likely support header values up to at least 8K, though, so we'd likely still be able to keep headers as the interface for clients to send this info.

Reply:
Matt Miller indicates that the payloads may be 300-400 bytes, but not as larger as 1024, so we should be fine to rely on Pubsub attributes.